pkg/cluster/internal/providers/podman/provision.go

```diff
@@ -307,6 +303,11 @@ func generatePortMappings(clusterIPFamily config.ClusterIPFamily, portMappings .
 		// generate the actual mapping arg
 		protocol := string(pm.Protocol)
 		hostPortBinding := net.JoinHostPort(pm.ListenAddress, fmt.Sprintf("%d", pm.HostPort))
+		// Podman expects empty string instead of 0 to assign a random port
+		// https://github.com/containers/libpod/blob/master/pkg/spec/ports.go#L68-L69
+		if strings.HasSuffix(hostPortBinding, ":0") {
```

Comment:
why don't check directly `pm.HostPort` before creating the `hostPortBinding` variable?

Reply:
That's not any less branchy and this is closer to the docker code /shrug

Reply:
fair enough